application/common/components/Emailer.php

```diff
@@ -410,24 +421,66 @@ public function sendDelayedMfaRelatedEmails()
     }
 
     /**
-     *
      * Whether the user has already been sent this type of email in the last X days
      *
      * @param int $userId
      * @param string $messageType
      * @return bool
      */
-    public function hasReceivedMessageRecently($userId, string $messageType)
+    public function hasUserReceivedMessageRecently($userId, string $messageType): bool
     {
-        $latestEmail = EmailLog::find()->where(['user_id' => $userId, 'message_type' => $messageType])
-            ->orderBy('sent_utc DESC')->one();
+        $user = User::findOne($userId);
+        if (empty($user)) {
+            \Yii::warning(sprintf(
+                'User (%s) not found when checking for recent emails',
+                $userId
+            ));
+            return false;
+        }
+        return $this->hasAddressReceivedMessageRecently($user->email, $messageType);
```

Comment:
Probably should use `getEmailAddress` rather than accessing the `email` property directly.
```suggestion
        return $this->hasAddressReceivedMessageRecently($user->getEmailAddress(), $messageType);
```

Reply:
Ahh, thanks. Good catch.